packages/client-proxy/internal/edge/sandboxes/catalog_redis.go

```diff
@@ -0,0 +1,121 @@
+package sandboxes
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/go-redsync/redsync/v4"
+	"github.com/jellydator/ttlcache/v3"
+	"github.com/redis/go-redis/v9"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	catalogClusterLockName = "sandboxes-catalog-cluster-lock"
+	catalogRedisTimeout    = time.Second * 5
+
+	// this is just how long we are keeping sandbox in local cache so we don't have to query redis every time
+	// we don't want to go too high because then sbx can be run on different orchestrator, and we will not be able to find it
+	catalogRedisLocalCacheTtl = time.Second * 10
```

Comment:
I think even 10 seconds is too much, also does it get updated in background or if I will be pinging it every 5 seconds, it will stay in the cache forever?